model/testing/src/icon4py/model/testing/pytest_config.py

```diff
@@ -78,20 +78,40 @@ def pytest_configure(config):
     if config.getoption("--backend"):
         backend_option = config.getoption("--backend")
         _check_backend_validity(backend_option)
+    if config.getoption("--datatest-only"):
+        k_option = config.getoption("-k")
+        if not k_option:
+            config.option.keyword = "datatest"
+        else:
+            new_option = f"({k_option}) and datatest"
+            config.option.keyword = new_option
```

Comment:
Optional suggestion:

```suggestion
        if (k_option := config.getoption("-k", [])):
            k_option = [f"({k_option})"]   # add parenthesis around original k_option just in case
        new_option = " and ".join(["datatest"] + k_option)
        config.option.keyword = new_option
```